crates/executor/src/simulate.rs

```diff
@@ -154,11 +181,14 @@ pub fn trace(
         let trace = to_trace(tx_type, tx_info, state_diff);
         traces.push((hash, trace));
     }
+
+    let _ = sender.send(traces.clone());
```

Comment:
What happens if we simply never get here (because of an error that happened earlier)?

Reply:
Hmm. Fair point, in my mind I thought all of these would end the same way so no harm. But ideally it should store the error as well? Or at least clear the cache item or something.

I think this probably needs more thought put into it. Panics etc are also near impossible to handle properly.

So we might need to place the computation on a separate thread? I think for now we just leave this out..

Reply:
Slightly related: https://fasterthanli.me/articles/request-coalescing-in-async-rust#panic-at-the-in-flight-request

Reply:
Why is the sender dropped here? I think it's still alive in the `CacheItem::Inflight` item set in the cache?

Reply:
Looks good to me. Have you tried sending a lot of concurrent requests for the same block?